src/python/WMCore/Services/pycurl_manager.py

```diff
@@ -78,6 +79,10 @@ def set_opts(self, curl, url, params, headers,
         curl.setopt(pycurl.FOLLOWLOCATION, self.followlocation)
         curl.setopt(pycurl.MAXREDIRS, self.maxredirs)
 
+        if params:
+            if  isinstance(params, dict):
+                params = json.dumps(params)
+
         if  verb == 'GET':
             encoded_data = urllib.urlencode(params, doseq=doseq)
```

Comment:
You're right, it's stressful day and my brain non working properly. I fixed the
code in PR.

On  0, ticoann <notifications@github.com> wrote:
> > @@ -78,6 +79,10 @@ def set_opts(self, curl, url, params, headers,
> >          curl.setopt(pycurl.FOLLOWLOCATION, self.followlocation)
> >          curl.setopt(pycurl.MAXREDIRS, self.maxredirs)
> >  
> > +        if params:
> > +            if  isinstance(params, dict):
> > +                params = json.dumps(params)
> > +
> >          if  verb == 'GET':
> >              encoded_data = urllib.urlencode(params, doseq=doseq)
> 
> Hi Valenitn, what I am saying is if the dictionary is passed for GET call then it is changed to json string urlencode will give the error.
> May be json dumps should be apply only non GET call?
> https://github.com/vkuznet/WMCore/blob/pycurl-fix/src/python/WMCore/Services/pycurl_manager.py#L91
> https://github.com/vkuznet/WMCore/blob/pycurl-fix/src/python/WMCore/Services/pycurl_manager.py#L95
> 
> ``` python
> >>> urllib.urlencode({'a':1, 'b':2})
> 'a=1&b=2'
> >>> urllib.urlencode(json.dumps({'a':1}))
> Traceback (most recent call last):
>   File "<stdin>", line 1, in <module>
>   File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib.py", line 1312, in urlencode
>     raise TypeError
> TypeError: not a valid non-string sequence or mapping object
> ```
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/dmwm/WMCore/pull/5868/files#r29908958